core/src/test/scala/unit/kafka/metrics/MetricsTest.scala

```diff
@@ -79,6 +79,17 @@ class MetricsTest extends KafkaServerTestHarness with Logging {
     assertFalse("Topic metrics exists after deleteTopic", checkTopicMetricsExists(topic))
   }
 
+  @Test
+  def testBrokerTopicMetricsAfterDeletingTopic() {
+    val topic = "test-broker-topic-metric"
+    AdminUtils.createTopic(zkUtils, topic, 2, 1)
+    createAndShutdownStep("group0", "consumer0", "producer0")
+    assertNotNull(BrokerTopicStats.getBrokerTopicStats(topic))
```

Comment:
`checkTopicMetricsExists` checks for any topic related metrics and that is good to check that all metrics related to the topic have been deleted. Here the check is for BrokerTopicMetrics specifically.